src/Microsoft.DotNet.Cli.Utils/Extensions/LockFileExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.DotNet.Tools.Common;
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    static class LockFileExtensions
+    {
+        public static string GetPackageDirectory(this LockFile lockFile, LockFileTargetLibrary library)
+        {
+            var packageFolders = lockFile.GetNormalizedPackageFolders();
+
+            var packageFoldersCount = packageFolders.Count();
+            var userPackageFolder = packageFoldersCount == 1 ? string.Empty : packageFolders.First();
```

Comment:
It did seem fragile to me but this is the logic that was [already there](https://github.com/dotnet/cli/blob/c72ed9fcd2d36eeb3354d3dcc0f78ddb6599bbc5/src/Microsoft.DotNet.Cli.Utils/CommandResolution/PackagedCommandSpecFactory.cs#L101-L106), I just moved it around.